src/main/java/com/github/matidominati/medicalclinic/service/PatientService.java

```diff
@@ -0,0 +1,75 @@
+package com.github.matidominati.medicalclinic.service;
+
+import com.github.matidominati.medicalclinic.model.Patient;
+import com.github.matidominati.medicalclinic.repository.PatientRepositoryImpl;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Optional;
+
+@Service
+@RequiredArgsConstructor
+public class PatientService {
+    private final List<Patient> patients;
+    private final PatientValidator patientValidator;
+    private final PatientRepositoryImpl patientRepository;
+
+    public List<Patient> getAllPatients() {
+        return patients;
+    }
+
+    public Patient getPatient(String email) {
+        Optional<Patient> patient = patientRepository.findPatientByEmail(email);
+        if (patient.isEmpty()) {
+            throw new RuntimeException("Patient with the provided email does not exist");
+        }
+        return patient.get();
+    }
+
+    public Patient addPatient(Patient patient) {
+        patientValidator.checkPatientData(patient);
+        if (patients.contains(patient)) {
+            throw new IllegalArgumentException("The patient already exists in the database");
+        }
+        patients.add(patient);
+        return patient;
+    }
+
+    public void deletePatient(String email) {
+        Optional<Patient> patientToDelete = patientRepository.findPatientByEmail(email);
```

Comment:
patientRepository.findPatientByEmail(email).orElseThrow(() -> throw new IllegalArgumentException("The patient with the given email address does not exist in the database"));